lib/services/northBound/contextServer.js

```diff
@@ -726,56 +1144,98 @@ function updateErrorHandling(error, req, res, next) {
     );
 }
 
+function updateErrorHandlingNgsi2(error, req, res, next) {
+    var code = 500;
+
+    logger.debug(context, 'Update NGSIv2 error [%s] handing request: %s', error.name, error.message);
+
+    if (error.code && String(error.code).match(/^[2345]\d\d$/)) {
+        code = error.code;
+    }
+
+    res.status(code).json({
+        error: error.name,
+        description: error.message.replace(/[<>\"\'=;\(\)]/g, '')
+    });
+}
+
 /**
  * Load the routes related to context dispatching (NGSI10 calls).
  *
  * @param {Object} router      Express request router object.
  */
 function loadContextRoutes(router) {
     //TODO: remove '//' paths when the appropriate patch comes to Orion
-    var updateMiddlewares = [
+    var updateMiddlewaresNgsi1 = [
             middlewares.ensureType,
-            middlewares.validateJson(updateContextTemplate),
-            handleUpdate,
-            updateErrorHandling
+            middlewares.validateJson(updateContextTemplateNgsi1),
+            handleUpdateNgsi1,
+            updateErrorHandlingNgsi1
         ],
-        queryMiddlewares = [
+        updateMiddlewaresNgsi2 = [
+            middlewares.ensureType,
+            middlewares.validateJson(updateContextTemplateNgsi2),
+            handleUpdateNgsi2,
+            updateErrorHandlingNgsi2
+        ],
+        queryMiddlewaresNgsi1 = [
             middlewares.ensureType,
             middlewares.validateJson(queryContextTemplate),
-            handleQuery,
-            queryErrorHandling
+            handleQueryNgsi1,
+            queryErrorHandlingNgsi1
+        ],
+        queryMiddlewaresNgsi2 = [
+            handleQueryNgsi2,
+            queryErrorHandlingNgsi2
         ],
-        updatePaths = [
+        updatePathsNgsi1 = [
             '/v1/updateContext',
             '/NGSI10/updateContext',
             '//updateContext'
         ],
-        queryPaths = [
+        updatePathsNgsi2 = [
+            '/v2/op/update',
+            '//op/update'
```

Comment:
In e2e CI I'm getting errors like "Cannot POST //op/query" and "Cannot POST //op/update". Looking to how it was defined in NGSIv1, it seems `//` (in addition to `/vX/`) is needed.

I don't fully understand this :) but let's do as in NGISv1 and see how it goes.

NTC (informative)